src/client.rs

```diff
@@ -34,27 +34,41 @@ pub struct Variant {
     pub name: String,
     pub payload: HashMap<String, String>,
     pub enabled: bool,
-}
-
-impl From<&CachedVariant> for Variant {
-    fn from(variant: &CachedVariant) -> Self {
-        Self {
-            name: variant.value.name.clone(),
-            payload: variant.value.payload.as_ref().cloned().unwrap_or_default(),
-            enabled: true,
-        }
-    }
+    pub feature_enabled: bool,
 }
 
 impl Variant {
-    fn disabled() -> Self {
+    fn disabled(feature_enabled: bool) -> Self {
```

Comment:
Fallout from the `feature_enabled` property in the client spec. A variant may be disabled because the feature is missing, in which case the feature is disabled, or it may be disabled because the feature has no variants, in which case it's up to the feature def whether or not its enabled